File: leetcode/editor/en/[880]Decoded String at Index.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var decodeAtIndex = function (s, k) {
    let size = BigInt(0);

    for (const c of s) {
        if (!isNaN(c)) {
            size *= BigInt(c);
        } else {
            size++;
        }
    }

    k = BigInt(k);

    for (let i = s.length - 1; i >= 0; i--) {
        const c = s[i];
        k %= size;

        if (!isNaN(c)) {
            size /= BigInt(c);
        } else {
            if (k === BigInt(0) || k === size) {
                return c;
            }

            size--;
        }
    }
    return "";
};
//leetcode submit region end(Prohibit modification and deletion)
/*
var decodeAtIndex = function (s, k) {
    let size = 0;
    let i = 0;

    while (size < k) {
        if (!isNaN(s[i])) {
            size *= Number(s[i]);
        } else {
            size++;
        }

        i++;
    }

    for (let j = i - 1; j >= 0; j--) {
        k %= size;
        if (!isNaN(s[j])) {
            size /= Number(s[j]);
        } else {
            if (k === 0 || k === size) {
                return s[j];
            }

            size--;
        }
    }
};
 */

File: editor/en/[880]Decoded String at Index.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public String decodeAtIndex(String s, int k) {
        long size = 0;

        for (char c : s.toCharArray()) {
            if (Character.isDigit(c)) {
                size *= c - '0';
            } else {
                size++;
            }
        }

        for (int i = s.length() - 1; i >= 0; i--) {
            char c = s.charAt(i);
            k %= size;

            if (Character.isDigit(c)) {
                size /= c - '0';
            } else {
                if (k == 0 || k == size) {
                    return String.valueOf(c);
                }

                size--;
            }
        }

        return "";
    }
}
//leetcode submit region end(Prohibit modification and deletion)
